source/extensions/filters/listener/proxy_protocol/proxy_protocol.cc

```diff
@@ -77,13 +77,16 @@ Network::FilterStatus Filter::onAccept(Network::ListenerFilterCallbacks& cb) {
 
 Network::FilterStatus Filter::onData(Network::ListenerFilterBuffer& buffer) {
   const ReadOrParseState read_state = parseBuffer(buffer);
-  if (read_state == ReadOrParseState::Error) {
+  switch (read_state) {
+  case ReadOrParseState::Error:
     config_->stats_.downstream_cx_proxy_proto_error_.inc();
     cb_->socket().ioHandle().close();
     return Network::FilterStatus::StopIteration;
-  } else if (read_state == ReadOrParseState::TryAgainLater) {
+  case ReadOrParseState::TryAgainLater:
     return Network::FilterStatus::StopIteration;
-  } else if (read_state == ReadOrParseState::SkipFilter) {
+  case ReadOrParseState::SkipFilter:
+    return Network::FilterStatus::Continue;
+  default:
```

Comment:
Please remove the default case. Then the compiler will warn/error if there are missing enum cases. I think it's possible that you won't need the `return` outside the switch either, if all cases are handled and all of them have a `return`.

Reply:
addressed in https://github.com/envoyproxy/envoy/pull/18951/commits/5c50e234f843f03e47eef8efda689899608ff139

unfortunately the compiler isn't smart enough to tell that if every case is implemented and returns, then we shouldn't need the ghost "default" case at the end of the function after the switch statement

```c++
source/extensions/filters/listener/proxy_protocol/proxy_protocol.cc: In member function 'virtual Envoy::Network::FilterStatus Envoy::Extensions::ListenerFilters::ProxyProtocol::Filter::onData(Envoy::Network::ListenerFilterBuffer&)':
source/extensions/filters/listener/proxy_protocol/proxy_protocol.cc:93:1: error: control reaches end of non-void function [-Werror=return-type]
   93 | }
      | ^
```

I removed the default case regardless and confirmed that compiler will fail if any case is not implemented